### PR TITLE
(FM-7764) - IOS Additional Syslog Settings created

### PR DESCRIPTION
**Pull request by david22swan**
- Attributes are:
	- `trap`, the syslog server logging level.
	- `origin_id`, an optional id to be added to all syslog messages.

### CODE REVIEW

**Comment by shermdog** [comment]
These are probably better suited as additions to netdev.  IOS and EOS both support `logging trap` and IOS and NXOS both support `logging origin-id`

File: README.md [hunk 1]
```diff
@@ -1042,6 +1043,55 @@ ID of the stp global config. Valid values are default.
...
+
+Examples:
+```
+ios_additional_syslog_settings { "default":
+  trap => 3,
+}
+```
+```
+ios_additional_syslog_settings { "default":
+  trap => 'unset,
```
[the remark this answers is not in the document]
Comment (david22swan) [reply README.md]: Fixed

File: README.md [hunk 2]
```diff
@@ -1042,6 +1043,55 @@ ID of the stp global config. Valid values are default.
...
+##### `origin_id`
+
+Data type: `Optional[Variant[Enum['hostname', 'ip', 'ipv6', unset], Tuple[Enum['string'], String]]]`
+
+Sets an origin-id to be added to all syslog messages, can be set to a default value taken from the switch itself or a designated one word string.
+
+Examples:
+```
+ios_additional_syslog_settings { "default":
+  origin_id => 'ipv6,
```
[the remark this answers is not in the document]
Comment (david22swan) [reply README.md]: fixed

File: README.md [hunk 3]
```diff
@@ -1042,6 +1043,55 @@ ID of the stp global config. Valid values are default.
...
+}
+```
+```
+ios_additional_syslog_settings { "default":
+  origin_id => ['string', 'Main'],
+}
+```
+```
+ios_additional_syslog_settings { "default":
+  origin_id => 'unset,
```
[the remark this answers is not in the document]
Comment (david22swan) [reply README.md]: fixed

File: lib/puppet/type/ios_additional_syslog_settings.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+require 'puppet/resource_api'
+
+Puppet::ResourceApi.register_type(
+  name: 'ios_additional_syslog_settings',
+  docs: 'Configure aaa accounting on device',
```
[the remark this answers is not in the document]
Comment (david22swan) [reply lib/puppet/type/ios_additional_syslog_settings.rb]: Fixed

File: lib/puppet/type/ios_additional_syslog_settings.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  features: ['remote_resource'],
+  attributes: {
+    name:         {
+      type:      'String',
+      desc:      'Name. On resource this is a composite of the authorization_service (and enable level if "commands") and authorization_list name eg. "commands 15 default" or "exec authlist1"',
```
[the remark this answers is not in the document]
Comment (david22swan) [reply lib/puppet/type/ios_additional_syslog_settings.rb]: Fixed

**Comment by david22swan** [comment]
additional unit test's added

**Pull request merged**